### DISCUSSION TITLE
Pretty URLs on `nix build`'ed static sites (`page/index.html` instead of `page.html`)

### DISCUSSION THREAD

**Discussion by omasanori** [srid/emanote#312] [Ideas]
Sorry if I overlooked something, but `urlStrategy: pretty` seems not working with `nix build`. Is it possible to use pretty URLs when building static sites?

**Comment by srid** [comment]
It should be under the "template" top level of index.yaml. See example: 

https://github.com/EmaApps/emanote/blob/master/docs/index.yaml

**Reply by omasanori** [reply]
Thank you for your prompt response!

Yes, I did and tried `emanote-template` as is. I am aware that routing `foo` to `foo.html` solves the problem but that configuration is not always available.

**Reply by srid** [reply]
@omasanori, I have trouble understanding the actual problem. Could you please produce a sequence of steps that I can run on my computer to reproduce your problem?

**Reply by srid** [reply]
For the record, you can always use `urlStrategy: direct` if you want explicit `.html` in URLs.

**Reply by omasanori** [reply]
OK.

1. Clone https://github.com/srid/emanote-template
2. `cd emanove-template`
3. Write something to `content/foo.md`, `git add` it and `git commit` it
4. Check if `urlStrategy` is pretty (it is the default of the template at that time)
5. `nix build`

Expected (or rather Desired) result: `foo/index.html` was generated

Actual result: `foo.html` was generated, a routing from `*` to `*.html` (by nginx's `try_file` or such) is needed

**Reply by srid** [reply]
Ah, it is in `nix build` generated site. You want to use pretty URLs in it?

It is currently hardcoded to use `urlDirect`: 

https://github.com/EmaApps/emanote/blob/074d0d3f96f5e35a6ca89da21a499fe5b70b75b2/nix/emanote.nix#L89

But we can add an `option` (see line 20) in this file to allow the user to configure it. If you are familiar with Nix, this would be a good first contribution for you to make :-)

**Reply by srid** [reply]
Oh, I missed one more thing:

> Expected (or rather Desired) result: foo/index.html was generated

`urlDirect` will generate `foo.html`. If you want `foo/index.html` generated instead, it would have to be done in Ema, and then used in Emanote. See https://github.com/EmaApps/ema/blob/67da7235a14e93e468112bb8d2217fb0e092304b/src/Ema/Route/Url.hs#L61-L67

EDIT: Nevermind! Actually, this change would belong to Emanote.

**Reply by omasanori** [reply]
Thank you! I will dig into the code...
